Automattic/jetpack
# Locale variants: allow switching from locale variant to the primary

#### Summary:
Due to the way the locale variant setting is stored in a separate option, switching from locale variant to the primary locale (e.g. from German (Formal) to German) didn't result in an update and nothing was saved.
This diff adds the check to fix it and modifies the API call to always return the `lang_id` setting as updated to account for that.

#### Testing instructions:
1. Checkout the diff and sandbox public-api
2. Go to https://wordpress.com/settings/general/, select a test site, set the language to English and save.
3. Change the site to the following languages, after every change go to different tab, then go back to settings, confirm that the language was changed, save it again without changing the language, then switch to a different tab and back to confirm that no 'You have unsaved changes' message would appear.

Deutsch
Deutsch (Sie)
Deutsch
Français
Deutsch (Sie)
Deutsch (Schweiz)
Italiano

Differential Revision: D61106-code

This commit syncs r225417-wpcom.

#### Does this pull request change what data or activity we track or use?
No
